pykokkos/lib/ufuncs.py

```diff
@@ -256,3 +256,491 @@ def sign(view):
+        view = pk.View([viewA.shape[0]], pk.float)
+        pk.parallel_for(
+            viewA.shape[0],
+            add_impl_1d_float,
+            view=view,
+            viewA=viewA,
+            viewB=viewB)
+
+    return view
+
+
+@pk.workunit
+def multiply_impl_1d_double(tid: int, view: pk.View1D[pk.double], viewA: pk.View1D[pk.double], viewB: pk.View1D[pk.double]):
+    view[tid] = viewA[tid] * viewB[tid]
+
+
+@pk.workunit
+def multiply_impl_1d_float(tid: int, view: pk.View1D[pk.float], viewA: pk.View1D[pk.float], viewB: pk.View1D[pk.float]):
+    view[tid] = viewA[tid] * viewB[tid]
+
+
+def multiply(viewA, viewB):
+    """
+    Multiplies positionally corresponding elements
+    of viewA with elements of viewB
+
+    Parameters
+    ----------
+    viewA : pykokkos view
+    viewB : pykokkos view
+
+    Returns
+    -------
+    view : pykokkos view
+    """
+
+    if str(viewA.dtype) == "DataType.double":
+        view = pk.View([viewA.shape[0]], pk.double)
+        pk.parallel_for(
+            viewA.shape[0],
+            multiply_impl_1d_double,
+            view=view,
+            viewA=viewA,
+            viewB=viewB)
+
+    elif str(viewA.dtype) == "DataType.float":
+        view = pk.View([viewA.shape[0]], pk.float)
+        pk.parallel_for(
+            viewA.shape[0],
+            multiply_impl_1d_float,
+            view=view,
+            viewA=viewA,
+            viewB=viewB)
+
+    return view
+
+
+@pk.workunit
+def subtract_impl_1d_double(tid: int, view: pk.View1D[pk.double], viewA: pk.View1D[pk.double], viewB: pk.View1D[pk.double]):
+    view[tid] = viewA[tid] - viewB[tid]
+
+
+@pk.workunit
+def subtract_impl_1d_float(tid: int, view: pk.View1D[pk.float], viewA: pk.View1D[pk.float], viewB: pk.View1D[pk.float]):
+    view[tid] = viewA[tid] - viewB[tid]
+
+
+def subtract(viewA, viewB):
+    """
+    Subtracts positionally corresponding elements
+    of viewA with elements of viewB
+
+    Parameters
+    ----------
+    viewA : pykokkos view
+    viewB : pykokkos view
+
+    Returns
+    -------
+    view : pykokkos view
+    """
+    if str(viewA.dtype) == "DataType.double":
+        view = pk.View([viewA.shape[0]], pk.double)
+        pk.parallel_for(
+            viewA.shape[0],
+            subtract_impl_1d_double,
+            view=view,
+            viewA=viewA,
+            viewB=viewB)
+
+    elif str(viewA.dtype) == "DataType.float":
+        view = pk.View([viewA.shape[0]], pk.float)
+        pk.parallel_for(
+            viewA.shape[0],
+            subtract_impl_1d_float,
+            view=view,
+            viewA=viewA,
+            viewB=viewB)
+
+    return view
+
+@pk.workunit
+def matmul_impl_1d_double(tid: int, acc: pk.Acc[pk.double], viewA: pk.View1D[pk.double], viewB: pk.View2D[pk.double]):
+    acc += viewA[tid] * viewB[0][tid]
+
+@pk.workunit
+def matmul_impl_1d_float(tid: int, acc: pk.Acc[pk.float], viewA: pk.View1D[pk.float], viewB: pk.View2D[pk.float]):
+    acc += viewA[tid] * viewB[0][tid]
```

Comment:
this is the naming thing I was talking about--`viewB` is clearly two dimensional here--maybe folks with a linalg background will have a better kernel name suggestion than `..1d..`, but may not matter for now. Or I guess you could argue it is a "column vector" that is conceptually 1D, dunno.

Reply:
I think these names are fine for now but matmul will definitely need to be revisited in the future for both this issue and performance